### PR TITLE
[bitnami/keycloak] Added ability to toggle http port on k8s service

### PR DESCRIPTION
**Pull request by andrewgkew**
Signed-off-by: Andrew Kew <andrew@quadcorps.co.uk>

### Description of the change

Added the ability to toggle off the http port on k8s service. 

### Benefits

This is useful when running keycloak behind a load balancer and only wanting https path to be accessible 

### Possible drawbacks

No drawbacks as you can leave the default value (true) and it works as before

### Applicable issues

None

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by andrewgkew** [comment]
@rafariossaa my PR keeps conflicting I assume when other merges bump the version (patch). I have bumped mine up a minor release version to  9.7.0 is this wrong?

File: bitnami/keycloak/Chart.yaml [hunk 1]
```diff
@@ -26,4 +26,4 @@ name: keycloak
 sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/keycloak
   - https://github.com/keycloak/keycloak
-version: 9.6.2
+version: 9.7.0
```
Comment (rafariossaa) [inline bitnami/keycloak/Chart.yaml:29]: Please, add a newline here.

**Review by rafariossaa — CHANGES_REQUESTED** [review]
Hi,
  This is ok. This repo has quite a bit of updates lately so sometimes you need to rebase several times you PR until it could be merged.

File: bitnami/keycloak/Chart.yaml (hunk 1 above)
Comment (andrewgkew) [reply bitnami/keycloak/Chart.yaml]: added

**Comment by andrewgkew** [comment]
@rafariossaa I will continue to fix the merge conflicts as the version change. Can you let me know what else you need from me to get this merged please. i have tested the logic locally and it works really well. Thanks

**Review by andrewgkew — COMMENTED** [review]
have added new line to chart file

**Comment by andrewgkew** [comment]
@joancafom just want to manage my expectation on when you think this PR might be looked at and hopefully merged :)

File: bitnami/keycloak/templates/service.yaml [hunk 2]
```diff
@@ -38,6 +38,7 @@ spec:
   sessionAffinityConfig: {{- include "common.tplvalues.render" (dict "value" .Values.service.sessionAffinityConfig "context" $) | nindent 4 }}
   {{- end }}
   ports:
+    {{- if .Values.service.http.enabled }}
```
[the remark this answers is not in the document]
Comment (joancafom) [reply bitnami/keycloak/templates/service.yaml:41]: You also need to replicate this in `headless-service.yaml`

**Review by joancafom — COMMENTED** [review]
Hi @andrewgkew 

Sorry for the delay! I have added some comments of your changes, could you please take a look at them when you have a chance?

File: bitnami/keycloak/templates/service.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewgkew) [reply bitnami/keycloak/templates/service.yaml:41]: @joancafom - I hear what you saying about the new field, but to me, the http port is by default set to port 80
To then turn it off you would need to set

`
service:
  ports:
    http: ""
`
I dont find that very intuitive. Having to set something to empty to turn it off?

Actually thinking about it using auth.tls.enabled for the https port seems a little weird too as its meant to relate to the service itself, so maybe we need to be more specific in the service config (now that I think about it)

`
service:
  http:
    enabled: true
    port: 80

  tls:
    enabled: true
    port: 443

`
What do you think? I am not sure about unsetting one parameter that can turn it off, I would rather be specific about it?

Re: headless-service I will update that ones we have come to a decision on the above.

**Comment by andrewgkew** [comment]
@marcosbc any movement on this PR?

**Review by marcosbc — APPROVED** [review]
LGTM! Thanks for the contribution! And sorry for the delay in checking this.

**Pull request merged**